src/transport/assembly_dg.hh

```diff
@@ -423,8 +453,21 @@ public:
                     sd[0] = s1; is_side_own[0] = edge_side1.cell().is_own();
                     sd[1] = s2; is_side_own[1] = edge_side2.cell().is_own();
 
-#define AVERAGE(i,k,side_id)  (fe_values_vec_[sd[side_id]].shape_value(i,k)*0.5)
-#define JUMP(i,k,side_id)     ((side_id==0?1:-1)*fe_values_vec_[sd[side_id]].shape_value(i,k))
+                    // precompute jumps and weighted averages of shape functions over the pair of sides (s1,s2)
+                    k=0;
+                    for (auto p1 : data_->stiffness_assembly_->edge_points(edge_side1) )
+                    {
+                        auto p2 = p1.point_on(edge_side2);
+                        for (unsigned int i=0; i<fe_->n_dofs(); i++)
+                        {
+                            for (int n=0; n<2; n++)
+                            {
+                                jumps[n][k*fe_->n_dofs()+i] = (n==0)*fe_values_vec_[s1].shape_value(i,k) - (n==1)*fe_values_vec_[s2].shape_value(i,k);
```

Comment:
Seems that jumps[0] == jumps[1], optimization possible.
We will further make an abstract storage for precomputed values for pairs of sides in order to hide exact placement in the memory.

Reply:
I don't think so. Shape functions can be ordered differently on opposite Edge sides.